File: src/main/java/com/bliss/blissapp/Service/QuoteService.java
package com.bliss.blissapp.Service;

import com.bliss.blissapp.Model.Quote;
import com.bliss.blissapp.Repository.QuoteRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.PathVariable;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class QuoteService {
    private final QuoteRepository quoteRepository;

    public List<Quote> getAllQuotes(){
        return quoteRepository.findAll();
    }

    public void createQuote(Quote quote) {
        quoteRepository.save(quote);
    }

    public Quote getRandomQuote() {
        List<Quote> allQuotes = getAllQuotes();
        int randomQuoteIndex = (int) Math.floor(Math.random() * (allQuotes.size()));
        return allQuotes.get(randomQuoteIndex);
    }
}

File: app/src/services/QuoteService.js
import axios from 'axios';

function quoteIdGenerator() {
    return Math.floor(Math.random() * (50) + 1);
}


class QuoteService {
    getQuote() {
        let QUOTE_REPO_URL = 'http://localhost:8080/quote/' + quoteIdGenerator();
        return axios.get(QUOTE_REPO_URL);
    }
}

export default new QuoteService()